authentik/sources/ldap/models.py

```diff
@@ -112,8 +123,20 @@ def server(self, **kwargs) -> Server:
         if self.peer_certificate:
             tls_kwargs["ca_certs_data"] = self.peer_certificate.certificate_data
             tls_kwargs["validate"] = CERT_REQUIRED
+        if self.client_certificate:
+            with NamedTemporaryFile(mode="w", delete=False) as temp_cert:
```

Comment:
I really wish there was a better way to do this than temporary files, we do something similar here https://github.com/goauthentik/authentik/blob/main/authentik/outposts/docker_tls.py, maybe it would make more sense to save a file named after the keypair into the temp folder so that it can be re-used, but a very small nitpick

Reply:
Yeah that probably also applies to the docker_tls wrapper, we could set better permission bits for the temporary file, but also from an attack vector, if someone has access to read those files they can do a lot of other things too